PixelHunter1995/WalkingAreaLib/PolygonPartition.cs

```diff
@@ -70,5 +71,38 @@ public void Draw(GraphicsDeviceManager graphics, SpriteBatch spriteBatch, double
         {
             Polygons.ForEach(p => p.Draw(graphics, spriteBatch, scaling, sceneWidth));
         }
+
+        public bool Contains(Vector2 position)
+        {
+            return Polygons.Any(polygon => polygon.Contains(position));
+        }
+
+        /// <summary>
+        /// Get next position to walk towards.
+        /// </summary>
+        public Vector2 GetNextPosition(Vector2 clickPosition, Vector2 currentPosition)
+        {
+            // clickPosition outside of PolygonPartition
+            if (!Contains(clickPosition))
+            {
+                // TODO: Get closest position on edge not just closest vertice
+                Vector2 closestVertice = Vector2.Zero;
+                double closestDistance = double.MaxValue;
+                foreach (var polygon in Polygons)
+                {
+                    (Vector2 closest, double distance) = polygon.GetClosestPosition(clickPosition);
```

Comment:
Don't the different polygons share corners? So we check every corner more than once?

Reply:
Yes , but will change a lot later anyway...